polkadot/runtime/parachains/src/scheduler.rs

```diff
@@ -190,7 +195,29 @@ pub mod pallet {
 	}
 }
 
-type PositionInClaimqueue = u32;
+type PositionInClaimQueue = u32;
+
+struct ClaimQueueIterator<E> {
+	next_idx: u32,
+	queue: Peekable<btree_map::IntoIter<CoreIndex, VecDeque<E>>>,
+}
+
+impl<E> Iterator for ClaimQueueIterator<E> {
```

Comment:
nit: IMHO it would have been easier to comprehend and code if instead of this custom iterator we'd just loop over the availability cores and try fetching the claimqueue for that core (in `fn eligible_paras`). still, this iterator has the slight advantage that when zipped with the av-cores, it'll stop at the end of the claim queue and not go further. feel free to ignore

Reply:
Also it allows for returning an iterator, otherwise you would return a closure with a reference that won't live long enough.